app/controllers/main/routes.py

```diff
@@ -47,18 +47,22 @@ def redirectToLogout():
 def landingPage():
 
     managerProgramDict = getManagerProgramDict(g.current_user)
-
     # Optimize the query to fetch programs with non-canceled, non-past events in the current term
+  
     programsWithEventsList = list(Program.select(Program, Event)
                                          .join(Event)
-                                         .where((Event.term == g.current_term) and (Event.isCanceled == False) and (Event.isPast == False))
+                                         .where((Event.term == g.current_term) & (Event.isCanceled == False))
                                          .distinct()
                                          .execute())  # Ensure only unique programs are included
+    
+    futureEvents = [p for p in programsWithEventsList if not p.event.isPast] #List Comprehension: filtered upcoming events from programWithEventsList 
```

Comment:
The comment should add clarity for future programmers about the intent of the developer or a step of an algorithm. Something like `# Limit returned list to events in the future` is better because it summarizes what the list comprehension is doing.

I also (usually) prefer comments on their own line before the code they are clarifying, so they can be read in-line with the code.